src/native/managed/cdac/mscordaccore_universal/Legacy/ClrDataAddress.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Diagnostics.DataContractReader.Legacy;
+
+/// <summary>
+/// Represents the native CLRDATA_ADDRESS 64-bit type which uses sign extending
+/// when converting from 32-bit values to 64-bit values.
+/// For more information see TO_CDADDR in dacimpl.h
+/// </summary>
+internal struct ClrDataAddress : IEquatable<ClrDataAddress>
```

Comment:
On some architectures, there are subtle calling convention differences between `PrimitiveType` and `struct WrapperOfPrimitiveType { PrimitiveType Value; }`, so the two are not interchangeable in portable code. I think it will probably work for this change since you only use it for `out` args.


ClrDataAddress